cmd/limactl/edit.go

```diff
@@ -53,14 +58,31 @@ func editAction(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	hdr := fmt.Sprintf("# Please edit the following configuration for Lima instance %q\n", instName)
-	hdr += "# and an empty file will abort the edit.\n"
-	hdr += "\n"
-	hdr += editutil.GenerateEditorWarningHeader()
-	yBytes, err := editutil.OpenEditor(instName, yContent, hdr)
+	tty, err := cmd.Flags().GetBool("tty")
 	if err != nil {
 		return err
 	}
+	yq, err := cmd.Flags().GetString("set")
+	if err != nil {
+		return err
+	}
+	var yBytes []byte
+	if yq != "" {
+		logrus.Warn("`--set` is experimental")
+		yBytes, err = yqutil.EvaluateExpression(yq, yContent)
+		if err != nil {
+			return err
+		}
+	} else if tty {
+		hdr := fmt.Sprintf("# Please edit the following configuration for Lima instance %q\n", instName)
+		hdr += "# and an empty file will abort the edit.\n"
+		hdr += "\n"
+		hdr += editutil.GenerateEditorWarningHeader()
+		yBytes, err = editutil.OpenEditor(instName, yBytes, hdr)
```

Comment:
`yBytes` is empty; we need to pass `yContent`:

```suggestion
		yBytes, err = editutil.OpenEditor(instName, yContent, hdr)
```

Reply:
My bad, it got lost when I redid/restructured. Previously the --set still worked with the editor